### PR TITLE
Cypress test to open all panes

### PR DESCRIPTION
**Pull request by RubenRBS**
This is a first draft as there are some things that are not properly implemented for which I need some help. Still, it should catch cases such as #3951

### CODE REVIEW

File: cypress/integration/frontend-testing.js [hunk 1]
```diff
@@ -1,23 +1,54 @@
...
+        });
+        // Can't bring the site to a default state by clearing localStorage, so do it manually
+        cy.get('[data-cy="more-dropdown-btn"]').click();
+        cy.get('[data-cy="reset-ui-btn"]:visible').click();
     });
 
-    runFrontendTest('HelloWorld');
-});
+    function addPaneOpenTest(datacy, title) {
+        it(title + " pane", () => {
```
Comment (RubenRBS) [inline cypress/integration/frontend-testing.js:27]: Had to move it directly to this file. I couldn't get the ts file in `static/test/` to accept `it`

File: tsconfig.frontend.json [hunk 2]
```diff
@@ -7,7 +7,8 @@
     /* Code generation */
     "inlineSources": true,
     "strictPropertyInitialization": false,
-    "lib": ["dom", "es5"]
+    "lib": ["dom", "es5"],
+    //"types": ["cypress"]
```
Comment (RubenRBS) [inline tsconfig.frontend.json:11]: This is me trying to get cypress to work in ts, but did nothing. Left it here to comment the try, but should remove before merging

**Review by RubenRBS — COMMENTED** [review]
Some comments from the implementation

**Comment by RubenRBS** [comment]
This passes locally for me, and indeed all the panes do run ok now except for the Graph one: https://github.com/compiler-explorer/compiler-explorer/runs/8170071588?check_suite_focus=true#step:5:419

**Comment by RubenRBS** [comment]
It finally works. Had to disable check on the Graph pane - For some reason it passes locally but not in CI. Will try to re-add once #3899 lands

File: cypress/integration/frontend-testing.js [hunk 3]
```diff
@@ -1,23 +1,119 @@
-function runFrontendTest(name) {
-    it(name, () => {
+import {it} from 'mocha';
+import {assertNoConsoleOutput, stubConsoleOutput} from '../support/utils';
+
+const PaneDataMap = {
```
Comment (matsjla) [inline cypress/integration/frontend-testing.js:4]: small nit; case style
```suggestion
const PANE_DATA_MAP = {
```

**Review by matsjla — APPROVED** [review]
This is great stuff! Found a tiny nit, but else LGTM

File: cypress/integration/frontend-testing.js [hunk 4]
```diff
@@ -1,23 +1,119 @@
...
+    addPaneOpenTest(PaneDataMap.device);
+    addPaneOpenTest(PaneDataMap.mir);
+    addPaneOpenTest(PaneDataMap.hir);
+    addPaneOpenTest(PaneDataMap.macro);
+    addPaneOpenTest(PaneDataMap.core);
+    addPaneOpenTest(PaneDataMap.stg);
+    addPaneOpenTest(PaneDataMap.cmm);
+    addPaneOpenTest(PaneDataMap.dump);
+    addPaneOpenTest(PaneDataMap.tree);
+    addPaneOpenTest(PaneDataMap.debug);
```
Comment (RubenRBS) [inline cypress/integration/frontend-testing.js:69]:
```suggestion
    addPaneOpenTest(PANE_DATA_MAP.executor);
    addPaneOpenTest(PANE_DATA_MAP.opt);
    addPaneOpenTest(PANE_DATA_MAP.preprocessor);
    addPaneOpenTest(PANE_DATA_MAP.ast);
    addPaneOpenTest(PANE_DATA_MAP.llvmir);
    addPaneOpenTest(PANE_DATA_MAP.pipeline);
    addPaneOpenTest(PANE_DATA_MAP.device);
    addPaneOpenTest(PANE_DATA_MAP.mir);
    addPaneOpenTest(PANE_DATA_MAP.hir);
    addPaneOpenTest(PANE_DATA_MAP.macro);
    addPaneOpenTest(PANE_DATA_MAP.core);
    addPaneOpenTest(PANE_DATA_MAP.stg);
    addPaneOpenTest(PANE_DATA_MAP.cmm);
    addPaneOpenTest(PANE_DATA_MAP.dump);
    addPaneOpenTest(PANE_DATA_MAP.tree);
    addPaneOpenTest(PANE_DATA_MAP.debug);
```

File: cypress/integration/frontend-testing.js [hunk 5]
```diff
@@ -1,23 +1,119 @@
...
+    afterEach('Ensure no output in console', () => {
+        cy.window().then(win => {
+            assertNoConsoleOutput();
+            win.localStorage.clear();
+        });
+    });
+
+    it('Correctly loads the page for a state with every pane active', () => {
+        for (const paneId in PaneDataMap) {
+            const pane = PaneDataMap[paneId];
```
Comment (RubenRBS) [inline cypress/integration/frontend-testing.js:111]:
```suggestion
        for (const paneId in PANE_DATA_MAP) {
            const pane = PANE_DATA_MAP[paneId];
```

**Review by RubenRBS — COMMENTED** [review]
opps

**Review by partouf — APPROVED** [review]

**Pull request merged**